[Swift in WebKit] Work towards proper WebGPU modularization (part 2)

rr-codes opened the pull request:
#### dbc483b29505781771b9c8335cdea707bf301b6d
<pre>
[Swift in WebKit] Work towards proper WebGPU modularization (part 2)
<a href="https://bugs.webkit.org/show_bug.cgi?id=302613">https://bugs.webkit.org/show_bug.cgi?id=302613</a>
<a href="https://rdar.apple.com/164853644">rdar://164853644</a>

Reviewed by Mike Wyrzykowski.

- All modules should have the `[system]` attribute. In this mode, `NSUInteger` is translated to Swift as `Int`.
Consequently, update some properties in `DDModelTypes` to stop using NSUInteger; for properties that just need
&quot;some reasonably large limited&quot; number, they should just use `NSInteger`, and properties that require a specific
width integer type should just use that type directly.

- All declarations in `DDModelTypes.h` are Objective-C, except for the enum declarations which are nonsensically
C++ enums instead of Objective C enums. Convert them to Objective C enums to be more consistent and substantially
easier to use in Swift.

* Source/WebGPU/WebGPU/DDMesh.mm:
(WebGPU::convert):
* Source/WebGPU/WebGPU/DDModelTypes.h:
* Source/WebGPU/WebGPU/Internal/module.modulemap:

Canonical link: <a href="https://commits.webkit.org/303258@main">https://commits.webkit.org/303258@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d2096e353cb48f61d46f56be2fad018c77e2ea5e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131717 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4209 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42717 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139226 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83582 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/5e3d5eab-7974-44d4-8004-308c8d0b25cc) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4237 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3971 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100687 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68119 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/949941b1-9761-4fcf-b669-f94f055d782a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134663 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3035 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117981 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81463 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/91c0c1c9-dcab-46b0-b7e1-c6909dd45efb) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2922 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/713 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82448 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111661 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141872 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3878 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36684 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3959 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3427 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27688 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2955 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114261 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57088 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3932 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32675 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3764 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67379 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4023 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3892 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (commented):
Some of these changes are fixed in https://github.com/WebKit/WebKit/pull/53644 or will otherwise not be needed

mwyrzykowski reviewed (approved):
This is great, there will still be conflicts with my PR but I can resolve them

rr-codes committed:
[Swift in WebKit] Work towards proper WebGPU modularization (part 2)
https://bugs.webkit.org/show_bug.cgi?id=302613
rdar://164853644

Reviewed by Mike Wyrzykowski.

- All modules should have the `[system]` attribute. In this mode, `NSUInteger` is translated to Swift as `Int`.
Consequently, update some properties in `DDModelTypes` to stop using NSUInteger; for properties that just need
"some reasonably large limited" number, they should just use `NSInteger`, and properties that require a specific
width integer type should just use that type directly.

- All declarations in `DDModelTypes.h` are Objective-C, except for the enum declarations which are nonsensically
C++ enums instead of Objective C enums. Convert them to Objective C enums to be more consistent and substantially
easier to use in Swift.

* Source/WebGPU/WebGPU/DDMesh.mm:
(WebGPU::convert):
* Source/WebGPU/WebGPU/DDModelTypes.h:
* Source/WebGPU/WebGPU/Internal/module.modulemap:

Canonical link: https://commits.webkit.org/303258@main
